src/ts/JsonEditorHelper.class.ts

```diff
@@ -122,62 +123,81 @@ export default class JsonEditorHelper {
   save(): void {
     if (this.editorDiv.dataset.what === 'file') {
       return this.saveFile();
-    } else if (this.editorDiv.dataset.what === 'metadata') {
+    }
+
+    if (this.editorDiv.dataset.what === 'metadata') {
       return this.saveMetadata();
     }
+
+    // toggle save menu so user can select what to save: file or metadata
+    if (this.editorDiv.dataset.what === '') {
+      if ($('#json-save-dropdown').next('div').is(':hidden')) {
+        $('#json-save-dropdown').dropdown('toggle');
+      }
+    }
+  }
+
+  // create a new file
+  saveNewFile(): void {
+    let realName = prompt(i18next.t('request-filename'));
+    if (realName === null) {
+      return;
+    }
+    // strip the filename of the .json extension from the name if available
+    if (realName.slice(-5).includes('.json')) {
+      realName = realName.slice(0, -5);
+    }
+    // add the new name for the file as a title
+    this.editorTitle.innerText = i18next.t('filename') + ': ' + realName + '.json';
+    $.post('app/controllers/EntityAjaxController.php', {
+      addFromString: true,
+      type: this.entity.type,
+      id: this.entity.id,
+      realName: realName,
+      fileType: 'json',
+      string: JSON.stringify(this.editor.get())
+    }).done(json => {
+      fetch(window.location.href).then(response => {
+        return response.text();
+      }).then(data => {
+        const parser = new DOMParser();
```

Comment:
with https://github.com/elabftw/elabftw/commit/6f6f1018824a93803c0661927c7ec41f91d92e15 I introduced `reloadElement()` so here you can just call `reloadElement('filesdiv')` ;)